network/p2p/connGater.go

```diff
@@ -13,28 +13,68 @@ import (
 
 var _ connmgr.ConnectionGater = (*ConnGater)(nil)
 
+// ConnGaterOption allow the connection gater to be configured with a list of PeerFilter funcs for a specific conn gater callback.
+// In the current implementation of the ConnGater the following callbacks can be configured with peer filters.
+// * InterceptPeerDial - peer filters can be configured with WithOnInterceptPeerDialFilters which will allow or disallow outbound connections.
+// * InterceptSecured - peer filters can be configured with WithOnInterceptSecuredFilters which will allow or disallow inbound connections after libP2P security handshake.
+type ConnGaterOption func(*ConnGater)
+
+// WithOnInterceptPeerDialFilters sets peer filters for outbound connections.
+func WithOnInterceptPeerDialFilters(filters []PeerFilter) ConnGaterOption {
+	return func(c *ConnGater) {
+		c.onInterceptPeerDialFilters = filters
+	}
+}
+
+// WithOnInterceptSecuredFilters sets peer filters for inbound secured connections.
+func WithOnInterceptSecuredFilters(filters []PeerFilter) ConnGaterOption {
+	return func(c *ConnGater) {
+		c.onInterceptSecuredFilters = filters
+	}
+}
+
 // ConnGater is the implementation of the libp2p connmgr.ConnectionGater interface
 // It provides node allowlisting by libp2p peer.ID which is derived from the node public networking key
 type ConnGater struct {
 	sync.RWMutex
-	peerFilter PeerFilter
-	log        zerolog.Logger
+	onInterceptPeerDialFilters []PeerFilter
+	onInterceptSecuredFilters  []PeerFilter
+	log                        zerolog.Logger
 }
 
-type PeerFilter func(peer.ID) bool
+type PeerFilter func(peer.ID) error
 
-func NewConnGater(log zerolog.Logger, peerFilter PeerFilter) *ConnGater {
+func NewConnGater(log zerolog.Logger, opts ...ConnGaterOption) *ConnGater {
 	cg := &ConnGater{
-		log:        log,
-		peerFilter: peerFilter,
+		log: log.With().Str("component", "connection_gater").Logger(),
+	}
+
+	for _, opt := range opts {
+		opt(cg)
 	}
 
 	return cg
 }
 
 // InterceptPeerDial - a callback which allows or disallows outbound connection
 func (c *ConnGater) InterceptPeerDial(p peer.ID) bool {
-	return c.peerFilter(p)
+	if len(c.onInterceptPeerDialFilters) == 0 {
+		c.log.Warn().
```

Comment:
Should this really be a `Warn`? Wouldn't Debug make more sense?

Reply:
https://github.com/onflow/flow-go/pull/2990/commits/c2fb003d4899db0bd8692210ffedbb4e95492ae4